### ISSUE TITLE
Question / suggestion. Host replacer modifier

### ISSUE THREAD

**Issue by mayrop** [thephpleague/uri#37]
I have been struggling with how to modify a whole host for a `uri`. I've come with the following code:

```
$uri = HttpUri::createFromString("http://www.example.com/to/the/sky/");
$modifier = new RemoveLabels(range(0, count($uri->host->keys()) - 2));
$newUri = $modifier->__invoke($uri);

$modifier = new ReplaceLabel(0, 'x.y.z');
echo $modifier->__invoke($newUri); // http://x.y.z/pat.s.xd.h/to/the/sky/
```

Which is little too wordy. In the past we were able to do:
`$uri->setHost($host);`

I've come something like (semicode):
```
class ReplaceHost extends AbstractHostModifier
{
    use Label;
    use Keys;

    /**
     * New instance
     *
     * @param string $label
     */
    public function __construct($label)
    {
        $this->label = $this->filterLabel($label);
    }

    /**
     * Modify a URI part
     *
     * @param string $str the URI part string representation
     *
     * @return string the modified URI part string representation
     */
    protected function modify($str)
    {
        $host = new Host($str);
        
        return (string) $host->without(range(0, count($host->keys()) - 1))->replace(0, $this->label);
    }
}
```

```
$modifier = new ReplaceHost('x.y.z');
echo $modifier->__invoke($uri);
```

If you like the idea, I can come with a pull request for replacing host and paths... since it is same annoying to replace a whole path too :(

**Comment by nyamsprod** [comment]
@mayrop thanks for using the League/uri package. The Uri class implements PSR-7 UriInterface so you are free to use the [basic manipulation methods](http://uri.thephpleague.com/uri/manipulation/#basic-modifications) explained in the documentation.

```php
$uri = HttpUri::createFromString("http://www.example.com/to/the/sky/");
$newUri = $uri->withHost('x.y.z');
echo $uri; // http://www.example.com/to/the/sky/
echo $newUri; // http://x.y.z/pat.s.xd.h/to/the/sky/
```

**Comment by mayrop** [comment]
I don't know how I missed that, going though all the other documentation. 

Sorry about it :(
